### PR TITLE
chore: add linters conforming to style guide

### PR DESCRIPTION
**Pull request by Xazin**
Relates: #56 #57

### CODE REVIEW

File: lib/src/flutter/overlay.dart [hunk 1]
```diff
@@ -338,8 +352,12 @@ class OverlayState extends State<Overlay> with TickerProviderStateMixin {
...
   int _insertionIndex(OverlayEntry? below, OverlayEntry? above) {
     assert(above == null || below == null);
-    if (below != null) return _entries.indexOf(below);
-    if (above != null) return _entries.indexOf(above) + 1;
+    if (below != null) {
+      return _entries.indexOf(below);
+    }
+    if (above != null) {
+      return _entries.indexOf(above) + 1;
+    }
```
Comment (Xazin) [inline lib/src/flutter/overlay.dart:360]:
```suggestion
    if (below != null) {
      return _entries.indexOf(below);
    } else if (above != null) {
      return _entries.indexOf(above) + 1;
    }
```

File: lib/src/flutter/overlay.dart [hunk 2]
```diff
@@ -430,25 +460,37 @@ class OverlayState extends State<Overlay> with TickerProviderStateMixin {
...
       'One or more of the specified entries are specified multiple times.',
     );
-    if (newEntriesList.isEmpty) return;
-    if (listEquals(_entries, newEntriesList)) return;
+    if (newEntriesList.isEmpty) {
+      return;
+    }
+    if (listEquals(_entries, newEntriesList)) {
+      return;
+    }
```
Comment (Xazin) [inline lib/src/flutter/overlay.dart:493]:
```suggestion
    if (newEntriesList.isEmpty || listEquals(_entries, newEntriesList)) {
      return;
    }
```

**Comment by LucasXu0** [comment]
Hey, @Xazin. There are too many lints that have been added. I'm concerned that there will be conflicts with other PRs if I merge this. How about we add only one lint for this PR?

**Comment by LucasXu0** [comment]
I believe that we should create a new issue to track each lint.

**Comment by Xazin** [comment]
> I believe that we should create a new issue to track each lint.

How about these lints, all changes are from `require_trailing_commas`, and around 10-15 are from the others.

**Comment by LucasXu0** [comment]
LGTM.

**Pull request merged**